baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/safety/SafetyAnnotations.java

```diff
@@ -205,6 +205,12 @@ private static Safety getSafetyInternal(Type type, VisitorState state, Set<Strin
             }
         }
         Safety typeArgumentCombination = SAFETY_IS_COMBINATION_OF_TYPE_ARGUMENTS.getSafety(type, state, dejaVu);
+        // Arrays are difficult to pipe through the above combiner without a large refactor, since the AST
+        // is not quite a tree it stores its type information adjacent to the node.
+        if (type instanceof Type.ArrayType) {
```

Comment:
IMPL based on feedback in the previous PR https://github.com/palantir/gradle-baseline/pull/2708#discussion_r1472018766